spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaOperation.java

```diff
@@ -98,6 +99,12 @@ public DeltaWriteBuilder newWriteBuilder(LogicalWriteInfo info) {
   public NamedReference[] requiredMetadataAttributes() {
     NamedReference specId = Expressions.column(MetadataColumns.SPEC_ID.name());
     NamedReference partition = Expressions.column(MetadataColumns.PARTITION_COLUMN_NAME);
+    if (TableUtil.formatVersion(table) >= 3) {
```

Comment:
instead of >= 3, maybe we want to move this to table util as well TableUtil.supportsRowLineage. It avoids the "magic format" number that's littered throughout now